jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -68,10 +68,16 @@ private void add(TarArchiveEntry tarArchiveEntry) {
         return;
       }
 
+      Path namePath = Paths.get(tarArchiveEntry.getName());
+
+      // Don't add entries for "/"
+      if (namePath.equals(Paths.get("/"))) {
```

Comment:
You said below that `TarArachiveEntry` strips a leading slash. For a root, does `tarArchiveEntry.getName()` returns `"/"` instead of `""` (empty string)? If it returns `""`, the check will never get matched, because `Paths.get("")` does not equal to `Paths.get("/")`.

Reply:
except for "/", there's special handling for "/", weirdly.

Reply:
Ah, interesting. Worth making a code comment.

Reply:
> except for "/", there's special handling for "/", weirdly.

Actually, I looked into the `TarArchiveEntry` code (`normalizeFileName`), and I think it strips the leading slash for the root too. I don't see any special handling for the root entry. And in my testing, it does return an empty string.

```java
        // No absolute pathnames
        // Windows (and Posix?) paths can start with "\\NetworkDrive\",
        // so we loop on starting /'s.
        while (!preserveAbsolutePath && fileName.startsWith("/")) {
            fileName = fileName.substring(1);
        }
        return fileName;
```

Reply:
I was looking more at: TarArchiveEntry.java:369 which occurs after normalization happens TarArchiveEntry.java:361.

For me, following the debugger puts it `/` when `tarArchiveEntry.getName()` is called.

Reply:
Also I think I see why we're getting different results, it looks like 
```
TarAchiveEntry("/").getName() != TarArchiveEntry(new File("some/file"), "/").getName()
```

Reply:
Ah, I see the difference. If you use `TarArchiveEntry(File)`, `getName()` returns `"/"`. However, when you use `TarArchiveEntry(String)`, it returns `""`.

Reply:
Anyway, there's obviously a lot to unpack here, lets just hold off on this PR completely till after we do a `2.1.0`.